src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -290,6 +333,37 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
     }
   }
 
+  void start_retry_block(long thread_id) {
+    std::unique_lock<std::mutex> lock(state_mutex);
+    auto thread = threads.find(thread_id);
+    if (thread != threads.end()) {
+      thread->second.reset_retry_state(true);
+    }
+  }
+
+  void end_retry_block(long thread_id) {
+    std::unique_lock<std::mutex> lock(state_mutex);
+    auto thread = threads.find(thread_id);
+    if (thread != threads.end()) {
+      thread->second.reset_retry_state(false);
+    }
+  }
+
+  long get_n_reset_lost_time(long task_id) {
```

Comment:
Nit: inconsistent use of `_n_` vs `_and_` in function names.  Would be nice to pick one and consistently use it.